csrc/tensor_view.cpp

```diff
@@ -1231,36 +1231,10 @@ TensorView* TensorView::cacheBefore(LoadStoreOpType cache_op) {
   // setDefinition(nullptr);
 
   auto replayed_consumer_pair =
-      TransformReplay::replayCasP(consumer, producer, -1);
+      TransformReplay::replayCasP(consumer, producer, -1, false, true);
 
   consumer->setDomain(replayed_consumer_pair.first);
 
-  // Recover allocation domain from transform replay
```

Comment:
This code is moved to `TransformReplay::replayCasP` so that it can be reused in `cacheAfter`.